library/data/registry-entry-attributes/src/main/java/org/quiltmc/qsl/registry/attribute/impl/RegistryEntryAttributeReloader.java

```diff
@@ -0,0 +1,226 @@
+import net.minecraft.util.Identifier;
+import net.minecraft.util.InvalidIdentifierException;
+import net.minecraft.util.JsonHelper;
+import net.minecraft.util.profiler.Profiler;
+import net.minecraft.util.registry.Registry;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+public final class RegistryEntryAttributeReloader implements SimpleResourceReloader<RegistryEntryAttributeReloader.LoadedData> {
+	private static final Logger LOGGER = LogManager.getLogger("AttributeReloader");
+	private static final Identifier ID = new Identifier("quilt", "attributes");
+
+	@Override
+	public Identifier getQuiltId() {
+		return ID;
+	}
+
+	@Override
+	public CompletableFuture<LoadedData> load(ResourceManager manager, Profiler profiler, Executor executor) {
+		return CompletableFuture.supplyAsync(() -> {
+			Map<RegistryEntryAttribute<?, ?>, AttributeMap> attributeMaps = new HashMap<>();
+
+			for (var entry : Registry.REGISTRIES.getEntries()) {
+				Identifier registryId = entry.getKey().getValue();
+				String path = getAttributeMapPath(registryId);
+				profiler.push(ID + "/finding_resources/" + path);
+
+				Collection<Identifier> jsonIds = manager.findResources("attributes/" + path, s -> s.endsWith(".json"));
+				if (jsonIds.isEmpty()) {
+					continue;
+				}
+
+				Registry<?> registry = entry.getValue();
+				BuiltinRegistryEntryAttributeHolder<?> holder = RegistryEntryAttributeHolderImpl.getBuiltin(registry);
+				processResources(manager, profiler, attributeMaps, jsonIds, registry, holder);
+
+				profiler.pop();
+			}
+
+			return new LoadedData(attributeMaps);
+		}, executor);
+	}
+
+	private void processResources(ResourceManager manager, Profiler profiler,
+								  Map<RegistryEntryAttribute<?, ?>, AttributeMap> attributeMaps,
+								  Collection<Identifier> jsonIds, Registry<?> registry,
+								  BuiltinRegistryEntryAttributeHolder<?> holder) {
+		for (var jsonId : jsonIds) {
+			Identifier attribId = getAttributeId(jsonId);
+			RegistryEntryAttribute<?, ?> attrib = holder.getAttribute(attribId);
+			if (attrib == null) {
+				LOGGER.warn("Unknown attribute {} (from {})", attribId, jsonId);
+				continue;
+			}
+
+			profiler.swap(ID + "/getting_resources{" + jsonId + "}");
+
+			List<Resource> resources;
+			try {
+				resources = manager.getAllResources(jsonId);
+			} catch (IOException e) {
+				LOGGER.error("Failed to get all resources for {}", jsonId);
+				LOGGER.catching(Level.ERROR, e);
+				continue;
+			}
+
+			profiler.swap(ID + "/processing_resources{" + jsonId + "," + attribId + "}");
+
+			AttributeMap attribMap = attributeMaps.computeIfAbsent(attrib,
+					id -> new AttributeMap(registry, attrib));
+			for (var resource : resources) {
+				attribMap.processResource(resource);
```

Comment:
This should probably catch all `IllegalArgumentException`s and `JsonParseExceptions`